tsdb/index/index_test.go

```diff
@@ -556,6 +556,242 @@ func TestReaderWithInvalidBuffer(t *testing.T) {
 	require.Error(t, err)
 }
 
+func TestReader_PostingsForMatcher(t *testing.T) {
+	testPostingsForMatcher(t, func(t *testing.T, series []labels.Labels) indexReader {
+		t.Helper()
+
+		dir := t.TempDir()
+		fn := filepath.Join(dir, indexFilename)
+		ctx := context.Background()
+
+		iw, err := NewWriter(ctx, fn)
+		require.NoError(t, err)
+
+		symbols := map[string]struct{}{}
+		for _, ls := range series {
+			ls.Range(func(l labels.Label) {
+				symbols[l.Name] = struct{}{}
+				symbols[l.Value] = struct{}{}
+			})
+		}
+
+		var syms []string
+		for s := range symbols {
+			syms = append(syms, s)
+		}
+		sort.Strings(syms)
+		for _, s := range syms {
+			require.NoError(t, iw.AddSymbol(s))
+		}
+
+		for i, s := range series {
+			require.NoError(t, iw.AddSeries(storage.SeriesRef(i), s))
+		}
+		require.NoError(t, iw.Close())
+
+		ir, err := NewFileReader(fn)
+		require.NoError(t, err)
+		t.Cleanup(func() {
+			require.NoError(t, ir.Close())
+		})
+
+		return blockIndexReader{
+			r: ir,
+		}
+	})
+}
+
+type blockIndexReader struct {
```

Comment:
Why "block" index reader?  What is block-ish about this?

Reply:
Removed.